crates/pathfinder/src/consensus/inner/persist_proposals.rs

```diff
@@ -158,3 +158,630 @@ impl<'tx> ConsensusProposals<'tx> {
         Ok(dto_block.into_model())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use p2p_proto::common::{Address, Hash, L1DataAvailabilityMode};
+    use p2p_proto::consensus::{BlockInfo, ProposalFin, ProposalInit};
+    use pathfinder_common::prelude::*;
+    use pathfinder_crypto::Felt;
+    use pathfinder_storage::StorageBuilder;
+
+    use super::*;
+
+    fn setup_test_db() -> (pathfinder_storage::Storage, pathfinder_storage::Connection) {
+        let storage = StorageBuilder::in_tempdir().expect("Failed to create temp database");
+        let mut conn = storage.connection().unwrap();
+        let tx = conn.transaction().unwrap();
+        tx.ensure_consensus_proposals_table_exists().unwrap();
+        tx.ensure_consensus_finalized_blocks_table_exists().unwrap();
+        tx.commit().unwrap();
+        (storage, conn)
+    }
+
+    fn create_test_proposal_parts(
```

Comment:
If you wish to go crazy with fake data you can use `fake::Dummy` to fill those parts with some random values. I'm not sure if it's needed though :thinking: .

Reply:
https://github.com/eqlabs/pathfinder/pull/3116/commits/837a13ff57da222640ba18fcb79c75bf70379d97